src/main/java/org/spongepowered/common/mixin/api/mcp/entity/monster/AbstractRaiderEntityMixin_API.java

```diff
@@ -42,4 +44,18 @@
         return Optional.ofNullable((Raid) this.shadow$getRaid());
     }
 
+    @Override
+    protected Set<Value.Immutable<?>> api$getVanillaValues() {
+        final Set<Value.Immutable<?>> vanillaValues = super.api$getVanillaValues();
+
+        vanillaValues.add(this.canJoinRaid().asImmutable());
+        vanillaValues.add(this.isCelebrating().asImmutable());
+
+        if (this.getRaid().isPresent()) { // A raid must be present for the raidWave to exist.
```

Comment:
Likewise don't need the if statement since the optional value will return empty.